### PR TITLE
[dev-v2.9] Bump chart cloud-provider-harvester to v0.2.3 in Rancher v2.9

### PR DESCRIPTION
**Pull request by w13915984028**
## Issue: 

https://github.com/rancher/rancher/issues/43838

## Problem

Bump chart cloud-provider-harvester for rancher version v2.9.

## Solution

Bump chart cloud-provider-harvester to v0.2.3

BTW, the `up0.2.3` fixed an important issue: https://github.com/harvester/harvester/issues/4678.

### CODE REVIEW

**Comment by nicholasSUSE** [comment]
Hello @w13915984028 
Is this Pull Request still relevant?
Can I close it?

**Comment by w13915984028** [comment]
@nicholasSUSE We still need this bump, I will rebase the PR to solve the conflict, thanks.

**Comment by w13915984028** [comment]
@nicholasSUSE @lucasmlp 

The PR is rebased, we have merged the https://github.com/rancher/charts/pull/3377 into v2.7.11; and this PR is for dev-v2.9, please help review and merge, thanks.

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/display-name: Harvester Cloud Provider
+  catalog.cattle.io/kube-version: '>= 1.23.0-0 < 1.27.0-0'
```
Comment (nicholasSUSE) [inline charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml]: Is this right? 
I think for Rancher 2.9+ the Kubernetes support should be higher am I wrong?

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  catalog.cattle.io/namespace: kube-system
+  catalog.cattle.io/os: linux
+  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.10.0-0'
```
Comment (nicholasSUSE) [inline charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml:7]: I am not very aware of Harvester but what I usually see is >= 2.7 < 2.10. 
Please correct me if I am wrong.

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  catalog.cattle.io/release-name: harvester-cloud-provider
+  catalog.cattle.io/ui-component: harvester-cloud-provider
+  catalog.cattle.io/upstream-version: 0.2.0
+apiVersion: v2
+appVersion: v0.2.0
```
Comment (nicholasSUSE) [inline charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml:12]: This is weird. 
```
appVersion: v0.2.0
version: 104.0.0+up0.2.3
```

Should it not be: `104.0.0+up0.2.0` OR `version: v0.2.3`?

**Review by nicholasSUSE — CHANGES_REQUESTED** [review]

**Comment by nicholasSUSE** [comment]
> @nicholasSUSE @lucasmlp
> 
> The PR is rebased, we have merged the #3377 into v2.7.11; and this PR is for dev-v2.9, please help review and merge, thanks.

@w13915984028 I have just reviewed, I am not very aware of `harvester` so I am looking more for patterns. 
If I made a mistake please correct me.

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml (hunk 1 above)
Comment (w13915984028) [reply charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml]: now updated to `1.23.0 - 1.30.0`
from the chart page, it is not claimed which k8s version is the upper, use 1.30.0 at the moment

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml (hunk 2 above)
Comment (w13915984028) [reply charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml:7]: Rancher-charts has branch dev-2.7, dev-2.8, dev-2.9, and the readme said the chart needs to be added to each branch manually.

From that, it seems we need to set the rancher version to be >= the current branch corresponding version.

File: charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml (hunk 3 above)
Comment (w13915984028) [reply charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml:12]: It is a mistake and is updated.

Comment (w13915984028) [reply charts/harvester-cloud-provider/104.0.0+up0.2.3/Chart.yaml:12]: appVersion is corresponding to the container image version
upstream-version is equal to upstream chart-version

The `harvester-cloud-provider` has different chart-verson and image version.

**Comment by w13915984028** [comment]
@nicholasSUSE Thanks for your review, pelase take another look of the updated PR.

BTW: When bumping a chart to a series of branchs like dev-v2.7, dev-v2.8, dev-v2.9, it needs to be much careful of all those minor differences.

**Review by nicholasSUSE — APPROVED** [review]
Looks good to me. @w13915984028

**Comment by Vicente-Cheng** [comment]
Hi @recena 

I am curious about the last time I did the bump rancher/charts.
Someone suggests splitting the bump commit into two commits.
1. the result of `make prepare` -> `make patch` -> `make clean`
2. the result of `make charts`

Do we still need to follow this?
(I thought that might be clear to check with reviewers, but I do not know what the rule is now)

**Comment by recena** [comment]
@Vicente-Cheng Added @rancher/mapps (team) as reviewer to give you advice.

**Comment by lucasmlp** [comment]
@Vicente-Cheng 

> Hi @recena
> 
> I am curious about the last time I did the bump rancher/charts. Someone suggests splitting the bump commit into two commits.
> 
> 1. the result of `make prepare` -> `make patch` -> `make clean`
> 2. the result of `make charts`
> 
> Do we still need to follow this? (I thought that might be clear to check with reviewers, but I do not know what the rule is now)

Yes, that is still the standard procedure. More info on this subject at [developing](https://github.com/rancher/charts/blob/dev-v2.9/docs/developing.md)

**Comment by Vicente-Cheng** [comment]
> @Vicente-Cheng
> 
> > Hi @recena
> > I am curious about the last time I did the bump rancher/charts. Someone suggests splitting the bump commit into two commits.
> > 
> > 1. the result of `make prepare` -> `make patch` -> `make clean`
> > 2. the result of `make charts`
> > 
> > Do we still need to follow this? (I thought that might be clear to check with reviewers, but I do not know what the rule is now)
> 
> Yes, that is still the standard procedure. More info on this subject at [developing](https://github.com/rancher/charts/blob/dev-v2.9/docs/developing.md)

Thanks, @lucasmlp,
So, @w13915984028 could we split this PR into two commits like your previous PR https://github.com/rancher/charts/pull/3377/commits ?

**Comment by w13915984028** [comment]
@Vicente-Cheng 
When we look into the commits of Rancher-charts, the commits are versatile.
Some are formed with 2 commits.
Some are 1 commit. e.g. https://github.com/rancher/charts/commit/f889aadf64fd63adffc7257c2fadb6c24489256a
Some are with more commits.

On the other hand, in practice, for this PR, it is rebased 10+ times, a single commit is easier to rebase.

Based on those, I would prefer keep the commit now. thanks.

**Comment by nicholasSUSE** [comment]
> @Vicente-Cheng When we look into the commits of Rancher-charts, the commits are versatile. Some are formed with 2 commits. Some are 1 commit. e.g. [f889aad](https://github.com/rancher/charts/commit/f889aadf64fd63adffc7257c2fadb6c24489256a) Some are with more commits.
> 
> On the other hand, in practice, for this PR, it is rebased 10+ times, a single commit is easier to rebase.
> 
> Based on those, I would prefer keep the commit now. thanks.

@w13915984028 
Thank you for your response. 
Indeed in the recent past, this rule has not been properly enforced and the documentation is unclear about it. 
It is fine for this PR to have 1 commit since you have rebased it a lot. 
Soon we will provide more clear documentation and automation for PR patterns.

**Review by Vicente-Cheng — APPROVED** [review]
Thanks @nicholasSUSE, @w13915984028 for the explanation.

LGTM, thanks!

**Pull request merged**